src/Microsoft.Health.Dicom.Blob/Features/Health/DicomConnectedStoreHealthCheck.cs

```diff
@@ -51,26 +52,22 @@ public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context
         _logger.LogInformation("Checking the health of the connected store.");
 
         BlobContainerClient containerClient = _blobClient.BlobContainerClient;
-        BlockBlobClient blockBlobClient = containerClient.GetBlockBlobClient($"{_externalStoreOptions.StorageDirectory}{_externalStoreOptions.HealthCheckFileName}");
+        BlockBlobClient healthCheckBlobClient = containerClient.GetBlockBlobClient(Path.Combine(_externalStoreOptions.StorageDirectory, $"{_externalStoreOptions.HealthCheckFilePath}{Guid.NewGuid()}.txt"));
 
-        // start trying to delete the blob, in case delete failed on the previous run
-        await TryDeleteBlob(blockBlobClient, cancellationToken);
```

Comment:
What about the old file? Should there be temp logic that deletes the old blob if it exists?

Reply:
Ah good catch - Added and put a comment with reference to the work item to clean up